st2client/st2client/commands/action.py

```diff
@@ -1495,70 +1500,113 @@ def tail_execution(cls, execution_manager, stream_manager, execution, output_typ
         execution = execution_manager.get_by_id(id=execution_id, **filters)
 
         children_execution_ids = getattr(execution, 'children', [])
-        workflow_execution_ids.update(children_execution_ids)
+        action_execution_ids.update(children_execution_ids)
 
+        workflow_execution_id = None
         events = ['st2.execution__update', 'st2.execution.output__create']
         for event in stream_manager.listen(events, **kwargs):
             status = event.get('status', None)
             is_execution_event = status is not None
 
             if is_execution_event:
                 context = cls.get_normalized_context_execution_task_event(event)
-                task_execution_id = context['execution_id']
+                action_execution_id = context['execution_id']
                 task_name = context['task_name']
-                task_parent_execution_id = context['parent_execution_id']
+                action_parent_execution_id = context['parent_execution_id']
+                task_id = context['task_execution_id']
+                wf_ex_id = context['workflow_execution_id']
+
+                if wf_ex_id and workflow_execution_id is None:
+                    workflow_execution_id = str(wf_ex_id)
+                    wf_ex_db = wf_db_access.WorkflowExecution.query(id=workflow_execution_id)[0]
```

Comment:
st2client is designed as a Python client library for StackStorm API which is also used by the CLI tool.

This means it can run anywhere and only communicates with StackStorm via the REST based HTTP API. This means we can't and shouldn't query the database here, but talk to the StackStorm API instead, if needed.

In addition to that, instead of retrieving those two objects from the API, we should see if we can augment the existing event stream events with this data which we need here.

Perhaps we could get it to work by also listening for the ``st2.workflow_execution__update`` and ``st2.task_execution__update`` event-stream events. For that to work though, we will need to update the code because, IIRC, we don't publish messages to the event bus for those objects at the moment (see ``st2common/st2common/persistance/execution.py`` for an example).

Right now this command utilizes StackStorm event stream API which means the client opens a long running connection to the stream API endpoint which "pushes" new events to this connection once they come in.

This would be preferred and a lot more efficient.

Performing 2 additional API calls for each event stream message would be simply too slow and expensive (imagine an action / workflow which outputs 10's of thousands of lines - in facts, that's pretty common with a lot of workflows).